cdap-app-fabric/src/main/java/co/cask/cdap/app/stream/DefaultStreamWriter.java

```diff
@@ -136,6 +136,7 @@ private void write(String stream, ByteBuffer data, Map<String, String> headers)
     for (Map.Entry<String, String> header : headers.entrySet()) {
       requestBuilder.addHeader(stream + "." + header.getKey(), header.getValue());
     }
+    addUserIdHeader(requestBuilder);
```

Comment:
That check was in the private method ``addUserIdHeader`` but I have now removed that method.